elastic/kibana
# [UserProfiles] Move React hook to package

In this PR I've moved the `useUpdateProfile()` hook from the security plugin start contract to the `kbn/user-profile-components` package. This hook was added in https://github.com/elastic/kibana/pull/161214 to be able to change the user theme color preference from within the avatar menu.

## How to test

* To test the dark mode toggle in the user menu, follow the "How to test" instruction in https://github.com/elastic/kibana/pull/161214
* To test the user profile page
  * Launch Kibana in classic mode
  * Click on the Avatar menu --> "Edit profile"
  * Change any user profile setting and verify that it is correctly saved

## Notes on implementations

* I've decided to expose the `UserProfileAvatarData`, `UserSettingsData` & `UserProfileData` from the package to keep a single interface for both the package and the `security` plugin

Fixes https://github.com/elastic/kibana/issues/161841
